circuits/circom/trees/incrementalMerkleTree.circom

```diff
@@ -104,10 +106,93 @@ template CheckRoot(levels) {
     // Initialize hashers for intermediate levels, each taking the outputs of two hashers from the previous level.
     var k = 0;
     for (var i = numLeafHashers; i < numLeafHashers + numIntermediateHashers; i++) {
-        computedLevelHashers[i] = PoseidonHasher(2)([hashers[k*2], hashers[k*2+1]]);
+        computedLevelHashers[i] = PoseidonHasher(2)([computedLevelHashers[k*2], computedLevelHashers[k*2+1]]);
         k++;
     }
 
     // Connect the output of the final hasher in the array to the root output signal.
     root <== computedLevelHashers[numHashers-1];
-}
\ No newline at end of file
+}
+
+/**
+ * Calculates the path indices required for Merkle proof verifications. 
+ * Given a node index within an IMT and the total tree levels, it outputs the path indices leading to that node.
+ * The template handles the modulo and division operations to break down the tree index into its constituent path indices.
+ */
+template MerkleGeneratePathIndices(levels) {
+    var BASE = 2;
+
+    signal input in; 
+    signal output out[levels];
+    signal n[levels + 1];
+
+    var m = in;
+    var computedResults[levels];
+
+    for (var i = 0; i < levels; i++) {
+        // circom's best practices suggests to avoid using <-- unless you
+        // are aware of what's going on. This is the only way to do modulo operation.
+        n[i] <-- m;        
+        out[i] <-- m % BASE;
+        m = m \ BASE;
+    }
+
+    n[levels] <-- m;
+
+    for (var i = 0; i < levels; i++) {
+        // Check that each output element is less than the base.
+        var computedIsOutputElementLessThanBase = SafeLessThan(3)([out[i], BASE]);
+        computedIsOutputElementLessThanBase === 1;
+
+        // Re-compute the total sum.
+        computedResults[i] = out[i] * (BASE ** i);
+    }
```

Comment:
Can we combine these two loops?

Reply:
let me do this in a separate PR, I saw that the one in `incrementalQuinTree` is the same and can be simplified as you suggested + remove unused variables

Reply:
Sure, it works

Reply:
opened issue here #1420